tests/preprocessing/test_imputation.py

```diff
@@ -20,17 +23,127 @@
 _TEST_PATH = f"{TEST_DATA_PATH}/imputation"
 
 
+def _base_check_imputation(
+    adata_before_imputation: AnnData,
+    adata_after_imputation: AnnData,
+    before_imputation_layer: str | None = None,
+    after_imputation_layer: str | None = None,
+    imputed_var_names: Iterable[str] | None = None,
+):
+    """Provides a base check for all imputations:
+
+    - Imputation doesn't leave any NaN behind
+    - Imputation doesn't modify anything in non-imputated columns (if the imputation on a subset was requested)
+    - Imputation doesn't modify any data that wasn't NaN
+
+    Args:
+        adata_before_imputation: AnnData before imputation
+        adata_after_imputation: AnnData after imputation
+        before_imputation_layer: Layer to consider in the original ``AnnData``, ``X`` if not specified
+        after_imputation_layer: Layer to consider in the imputated ``AnnData``, ``X`` if not specified
+        imputed_var_names: Names of the features that were imputated, will consider all of them if not specified
+
+    Raises:
+        AssertionError: If any of the checks fail.
+    """
+
+    layer_before = _to_dense_matrix(adata_before_imputation, before_imputation_layer)
+    layer_after = _to_dense_matrix(adata_after_imputation, after_imputation_layer)
+
+    if layer_before.shape != layer_after.shape:
+        raise AssertionError("The shapes of the two layers do not match")
+
+    var_indices = (
+        np.arange(layer_before.shape[1])
+        if imputed_var_names is None
+        else [
+            adata_before_imputation.var_names.get_loc(var_name)
+            for var_name in imputed_var_names
+            if var_name in imputed_var_names
+        ]
+    )
+
+    before_nan_mask = _is_val_missing(layer_before)
+    imputed_mask = np.zeros(layer_before.shape[1], dtype=bool)
+    imputed_mask[var_indices] = True
+
+    # Ensure no NaN remains in the imputed columns of layer_after
+    if np.any(before_nan_mask[:, imputed_mask] & _is_val_missing(layer_after[:, imputed_mask])):
+        raise AssertionError("NaN found in imputed columns of layer_after.")
+
+    # Ensure unchanged values outside imputed columns
+    unchanged_mask = ~imputed_mask
+    if not _are_ndarrays_equal(layer_before[:, unchanged_mask], layer_after[:, unchanged_mask]):
+        raise AssertionError("Values outside imputed columns were modified.")
+
+    # Ensure imputation does not alter non-NaN values in the imputed columns
+    imputed_non_nan_mask = (~before_nan_mask) & imputed_mask
+    if not _are_ndarrays_equal(layer_before[imputed_non_nan_mask], layer_after[imputed_non_nan_mask]):
+        raise AssertionError("Non-NaN values in imputed columns were modified.")
+
+    # All checks passed
```

Comment:
```suggestion
```

Reply:
I think that's self evident enough but this is a nitpick! You don't have to remove it.